staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/structuredmerge.go

```diff
@@ -141,7 +141,9 @@ func (f *structuredMergeManager) Apply(liveObj, patchObj runtime.Object, managed
 
 	patchObjTyped, err := f.typeConverter.ObjectToTyped(patchObj)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create typed patch object (%v): %v", objectGVKNN(patchObj), err)
+		return nil, nil, errors.NewBadRequest(
```

Comment:
is this always a bad request or could it be an internal error? I can't tell where the fault lies if this encounters an error

Reply:
this also needs a test (probably an integration test) demonstrating the issue being fixed